### PR TITLE
K8SPG-42 Remove podName from backup task

### PR DESCRIPTION
**Pull request by nonemax**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| scaling | passed |
| recreate | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/45/commits/86d26607ef14314e304c5a29bd011f26497cb29d
image: `perconalab/percona-postgresql-operator:PR-45-86d26607-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-45-86d26607-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-45-86d26607-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-45-86d26607-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-45-86d26607-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-45-86d26607-pgo-deployer`

**Review by qJkee — APPROVED** [review]

File: internal/operator/backrest/backup.go [hunk 1]
```diff
@@ -97,12 +97,25 @@ func Backrest(namespace string, clientset kubeapi.Interface, task *crv1.Pgtask)
...
+		FieldSelector: fields.OneTermEqualSelector("status.phase", string(v1.PodRunning)).String(),
+		LabelSelector: selector,
+	}
 
+	pods, err := clientset.CoreV1().Pods(cluster.Namespace).List(ctx, options)
+	if err != nil {
+		log.Errorf("error getting pods from selector %s: %s", selector, err)
+		return
+	} else if len(pods.Items) != 1 {
+		log.Error("pods returned does not equal 1, it should")
```
Comment (cap1984) [inline internal/operator/backrest/backup.go:111]: Please rewrite it to human readable message)

Comment (nonemax) [reply internal/operator/backrest/backup.go]: done in https://github.com/percona/percona-postgresql-operator/pull/45/commits/0a99c65b1a6637330ccb6eddbf005fd642976506

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| scaling | passed |
| recreate | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/45/commits/0a99c65b1a6637330ccb6eddbf005fd642976506
image: `perconalab/percona-postgresql-operator:PR-45-0a99c65b-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-45-0a99c65b-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-45-0a99c65b-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-45-0a99c65b-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-45-0a99c65b-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-45-0a99c65b-pgo-deployer`

File: deploy/operator.yaml [hunk 2]
```diff
@@ -180,7 +180,7 @@ data:
     pgo_image_pull_policy: "Always"
     pgo_image_pull_secret: ""
     pgo_image_pull_secret_manifest: ""
-    pgo_image_tag: "main"
+    pgo_image_tag: "k8spg-42"
```
Comment (cap1984) [inline deploy/operator.yaml:183]: really?)

Comment (nonemax) [reply deploy/operator.yaml]: no) https://github.com/percona/percona-postgresql-operator/pull/45/commits/b05e9012d113a41b342e30296f1279f5113c7b6f

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| scaling | passed |
| recreate | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/45/commits/b05e9012d113a41b342e30296f1279f5113c7b6f
image: `perconalab/percona-postgresql-operator:PR-45-b05e9012-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-45-b05e9012-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-45-b05e9012-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-45-b05e9012-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-45-b05e9012-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-45-b05e9012-pgo-deployer`

File: cmd/pgo-scheduler/scheduler/pgbackrest.go [hunk 3]
```diff
@@ -136,7 +136,7 @@ func (b BackRestBackupJob) Run() {
 			"selector":  selector,
 			"error":     err,
 			"podsFound": len(pods.Items),
-		}).Error("pods returned does not equal 1, it should")
+		}).Error("error getting pods from selector: 0 or more then 1 pods present")
```
Comment (qJkee) [inline cmd/pgo-scheduler/scheduler/pgbackrest.go:139]:
```suggestion
		}).Error("error getting pods from selector: 0 or more than 1 pods present")
```

Comment (hors) [reply cmd/pgo-scheduler/scheduler/pgbackrest.go:139]: @nonemax please fix it and check all messages too

**Review by hors — CHANGES_REQUESTED** [review]

File: cmd/pgo-scheduler/scheduler/pgbackrest.go (hunk 3 above)
Comment (nonemax) [reply cmd/pgo-scheduler/scheduler/pgbackrest.go]: fixed https://github.com/percona/percona-postgresql-operator/pull/45/commits/f1cc9b16e769c3f45b077f836a339f1be7bf1f9f

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| scaling | passed |
| recreate | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/45/commits/4e6ba8b803b2e0d9de97f3b34e7e8665c7534b3f
image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| scaling | passed |
| recreate | failed |

commit: https://github.com/percona/percona-postgresql-operator/pull/45/commits/f1cc9b16e769c3f45b077f836a339f1be7bf1f9f
image: `perconalab/percona-postgresql-operator:PR-45-f1cc9b16-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-45-f1cc9b16-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-45-f1cc9b16-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-45-f1cc9b16-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-45-f1cc9b16-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-45-f1cc9b16-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-postgresql-operator/pull/45/commits/4e6ba8b803b2e0d9de97f3b34e7e8665c7534b3f
image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-deployer`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| scaling | passed |
| recreate | passed |

commit: https://github.com/percona/percona-postgresql-operator/pull/45/commits/4e6ba8b803b2e0d9de97f3b34e7e8665c7534b3f
image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-apiserver`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-event`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-rmdata`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-scheduler`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-postgres-operator`

image: `perconalab/percona-postgresql-operator:PR-45-4e6ba8b8-pgo-deployer`

**Review by hors — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Pull request merged**